spec/tests/restart-test-on-exit-code.fmf

```diff
@@ -0,0 +1,53 @@
+summary: Handling test crashes
+
+story: |
+    As a tester I want to run tests that may trigger kernel panic or
+    cripple their guest in other ways, often on purpose. I need a way to
+    reliably continue with the testing process.
+
+description: |
+    Some tests may focus on lower levels of system functionality, and
+    perform actions that cause system crashes. And the crashes might be
+    triggered on purpose, e.g. to verify a system can recover.
+
+    tmt on its own cannot detect a kernel panic, and cannot pick from
+    all possible ways of handling such a situation, therefore offers
+    tests a way to hint tmt on how to proceed:
+
+    See :ref:`plugins/test-checks/watchdog` for a test-level check that
+    can detect frozen guests and trigger hard reboot before restarting
+    the test.
+
+    restart-test-on-exit-code: ``EXIT-CODES``
+        When set, it lists test exit codes that should trigger the test
+        restart.
+
+        Default: *not set*
+
+    restart-test-max-times: ``LIMIT``
+        How many times the test may be restarted before giving up. It
+        must be at least 1, and the upper limit is 10.
+
+        Default: ``1``
+
+    reboot-before-test-restart: ``true|false``
+        When set, a hard reboot would be triggered before restarting the
+        test.
```

Comment:
Several thoughts around the naming:

* Would it make sense to use a common prefix for all as they are very closely related?
* All of them are related to test, so could we drop the `test` part from the names?

What about this?

* `restart-exit-code`
* `restart-max-count`
* `restart-with-reboot`

I see that explicitly mentioning `test` makes it more clear that it's not about the system restart, on the other hand the names seem a little bit too long. With the full common prefix it could look like this:

* `restart-test-exit-code`
* `restart-test-max-count`
* `restart-test-with-reboot`

Thoughts?

Reply:
I would be fine with these.

Let me know which set of names should I use. I don't want to jump into the editor right away & replace everything with these just to see a comment a few minutes later disputing their sanity.

Reply:
One note though: `restart-test-exit-code` drops the `on` bit, and suddenly it's less clear the restart of test is triggered by these exit codes. `restart-test-exit-code` sounds more like it's an exit code tmt should somehow produce when test restarts, the cause & effect direction is less clear to me.

Reply:
Understood. Then let's go with these:

* `restart-on-exit-code`
* `restart-max-count`
* `restart-with-reboot`

Reply:
Renamed in dec2be5c